folium/folium.py

```diff
@@ -538,55 +556,73 @@ def choropleth(self, geo_data, data=None, columns=None, key_on=None,
         else:
             color_data = None
 
-        # Compute color_domain
-        if threshold_scale is not None:
-            color_domain = list(threshold_scale)
-        elif color_data:
-            # To avoid explicit pandas dependency ; changed default behavior.
-            data_min = min(color_data.values())
-            data_max = max(color_data.values())
-            if data_min == data_max:
-                data_min = (data_min if data_min < 0 else 0
-                            if data_min > 0 else -1)
-                data_max = (data_max if data_max > 0 else 0
-                            if data_max < 0 else 1)
-            data_min, data_max = (1.01*data_min-0.01*data_max,
-                                  1.01*data_max-0.01*data_min)
-            nb_class = 6
-            color_domain = [data_min+i*(data_max-data_min)*1./nb_class
-                            for i in range(1+nb_class)]
-        else:
-            color_domain = None
+        if color_data is not None and key_on is not None:
+            # numpy histogram_bin_edges is only available in the most
+            # recent numpy version, so we will use numpy histogram
+            real_values = np.array(list(color_data.values()))
+            real_values = real_values[~np.isnan(real_values)]
+            _, bins = np.histogram(real_values, bins=bins)
```

Comment:
I'm not a fan of reusing variable names when their type may have changed. Can we use a different name? In Numpy they name the output `bin_edges`.

Reply:
I went for `bin_edges`